Mu2eG4/src/constructPSEnclosure.cc

```diff
@@ -91,101 +103,275 @@ namespace mu2e {
+                 findMaterialOrThrow(pse->windows()[i].materialName()),
+                 0,
+                 pse->windows()[i].originInMu2e() - parent.centerInMu2e() + extraOffset,
+                 parent,
+                 0,
+                 PSIsVisible,
+                 G4Colour::Grey(),
+                 PSIsSolid,
+                 forceAuxEdgeVisible,
+                 placePV,
+                 doSurfaceCheck
+                 );
+
+
+        verbosityLevel
+          && std::cout << __func__ << " " << wname.str() << " center in Mu2e: "
+                       <<pse->windows()[i].originInMu2e()
+                       << std::endl;
+
+	//      verbosityLevel
+	//        && std::cout << __func__ << " " << sName << " Mass in kg: "
+	//                     << _helper->locateVolInfo(sName).logical->GetMass()/CLHEP::kg
+	//                     << std::endl;
+
+
+      } //end window for loop
+
+    } //end version < 3 endplate construction
+
+    else { //build version >= 3 endcap
+      auto polycone = pse->endPlatePolycone();
       verbosityLevel
-        && std::cout << __func__ << " " << wname.str() << " center in Mu2e: "
-                     <<pse->windows()[i].originInMu2e()
+        && std::cout << __func__ << ": PSEndPlate polycone: "
+                     << polycone
                      << std::endl;
-    }
+      G4Polycone* basecone = new G4Polycone("PSEnclosureEndplateBaseCone",
+                                            polycone.phi0(),
+                                            polycone.phiTotal(),
+                                            polycone.numZPlanes(),
+                                            &polycone.zPlanes()[0],
+                                            &polycone.rInner()[0],
+                                            &polycone.rOuter()[0]
+                                            );
+      G4VSolid* solid = basecone;
+      // double zlength = 0.;
+      // if(polycone.zPlanes().size() > 0 ) {
+      //   zlength = abs(polycone.zPlanes()[0] - polycone.zPlanes()[polycone.zPlanes().size() - 1]); //planes should be ordered
+      // }
+      CLHEP::Hep3Vector coneOrigin = polycone.originInMu2e();
+      // coneOrigin.setZ(coneOrigin.z());
+      //punch a hole in the endcap for each window
+      for(unsigned iwindow = 0; iwindow<pse->nWindows(); ++iwindow) {
+        Tube pipe = pse->windowPipes()[iwindow];
+        std::ostringstream pipename, solidname, windowname, boxname;
+        pipename << "PSEnclosurePipeHole_" << iwindow+1;
+        solidname << "PSEnclosurePlateTmp_" << iwindow+1;
+        boxname   << "PSEnclosurePlateBox_" << iwindow+1;
+        windowname   << "PSEnclosureWindow_" << iwindow+1;
+        //make a solid pipe that is the hole in the pipe, to make a hole in the endplate
+        G4Tubs* pipetube = new G4Tubs(pipename.str(), 0., pipe.innerRadius(), pipe.halfLength(), 0., CLHEP::twopi);
+        G4RotationMatrix* matrix = new G4RotationMatrix(pipe.rotation());
```

Comment:
new G4RotationMatrix(pipe.rotation()) is a memory leak.
Please use AntiLeakRegistry registry as e.g., in constructPSShield and many other places 
(and perhaps use CLHEP::HepRotation instead of the typedefed G4RotationMatrix for consistency with the majority of the Mu2eG4 code)

Reply:
I've added this to the AntiLeakRegistry, thank you for noticing this!